### PR TITLE
Add_aws_endpoint_url_var

### PR DESCRIPTION
**Pull request by lakkeger**
# Motivation
Add support for AWS_ENDPOINT_URL env variable.

# Changes
- Add AWS_ENDPOINT_URL env variable
- Small improvements

### CODE REVIEW

File: bin/pulumilocal [hunk 1]
```diff
@@ -14,18 +14,20 @@ import argparse
...
 
 # define global constants
-TRUE_STRINGS = ['1', 'true', 'True']
+TRUE_STRINGS = ['1', 'true']
 PULUMI_CMD = os.environ.get('PULUMI_CMD') or 'pulumi'
-LOCALSTACK_HOSTNAME = os.environ.get('LOCALSTACK_HOSTNAME') or 'localhost'
-EDGE_PORT = int(os.environ.get('EDGE_PORT') or '4566')
-USE_SSL = os.environ.get('USE_SSL') in TRUE_STRINGS
+AWS_ENDPOINT_URL = os.environ.get("AWS_ENDPOINT_URL")
+LOCALSTACK_HOSTNAME = urlparse(os.environ.get('AWS_ENDPOINT_URL')).hostname or os.environ.get("LOCALSTACK_HOSTNAME") or 'localhost'
```
Comment (whummer) [inline bin/pulumilocal]: Wasn't aware that `urlparse(None)` works, but just quickly tested locally and doesn't raise an error 👍 

Small nit: would be good to have uniform style for quotes (single/double quotes) - maybe we can adjust the linter settings to use `black`/`ruff` for the `make lint` target, as we have it in our other repos (as opposed to `flake8`).

**Review by whummer — APPROVED** [review]
Great set of changes @lakkeger - only a small nitpick suggestion. 👍

File: bin/pulumilocal (hunk 1 above)
Comment (lakkeger) [reply bin/pulumilocal]: Absolutely. In a following patch I'll add a different linter (probably ruff) and introduce pre-commit hooks, until that I fixed manually the quotation mark inconsistency.

**Comment by joe4dev** [comment]
same comment here @lakkeger :)

It would be great if you could synchronize the documentation here accordingly: https://github.com/localstack/docs/blob/23f5bdf1eba5b2a04716ebc57cb6e2364b0c8f06/content/en/user-guide/integrations/pulumi/index.md?plain=1#L456C1-L458

**Pull request merged**